framework/src/problems/FEProblemBase.C

```diff
@@ -565,6 +569,15 @@ FEProblemBase::createTagVectors()
       _nl[nl_sys_num]->addVector(tag, false, GHOSTED);
     }
 
+  auto & not_zeroed_vectors = getParam<std::vector<std::vector<TagName>>>("not_zeroed_tag_vectors");
+  for (const auto nl_sys_num : index_range(not_zeroed_vectors))
+    for (auto & vector : not_zeroed_vectors[nl_sys_num])
+    {
+      auto tag = addVectorTag(vector);
+      _nl[nl_sys_num]->addVector(tag, false, GHOSTED);
```

Comment:
```suggestion
      _nl[nl_sys_num]->addVector(tag, false, PARALLEL);
```
If you're feeling brave I would try changing this on line 569 as well

Reply:
If you wouldn't mind explaining, for my own knowledge, what is the difference between these two settings?

Reply:
It's annoying that github hid this conversation even though it wasn't resolved. If a vector is ghosted, then you can read entries in the vector that are not owned by your processor. It would have been nice if I had told you this before you ran into your issues below!!

Reply:
As noted in https://github.com/idaholab/moose/pull/28177#issuecomment-2303804746 let's revert my suggestion. My apologies!